### PR TITLE
[Accessibility] Add scope attributes to the order table on My Account

### LINKED ISSUE #43629: Accessibility: My Account orders table should have row-scoped headers in addition to column headers
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

The orders table on the My Account page currently only has headers for the columns. The Order number (first column) also functions as a header and should be in a TH tag with a scope of row rather than a TD tag.

Relevant code:
https://github.com/woocommerce/woocommerce/blob/758df4854d7b627fc26eb5c0860dc492dea6a06e/plugins/woocommerce/templates/myaccount/orders.php#L26

### Expected behavior

1. All of the column headers need scope="col" added to them.
2. Change the order number cell to a `<th> `with scope="row".

This will allow screen readers to more clearly connect the order number with the other cells in the table.

### Actual behavior

Order numbers are not clearly connected to the date, status, total, and actions.

### Steps to reproduce

Inspect the orders table or see code linked above.

### WordPress Environment

N/A

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by Manussakis**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR adds the `scope` HTML attribute to improve the accessibility of the orders table on the My Account page.
Closes #43629 .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

This PR adds more semantics to the Orders table.
The following steps to QA consider the test with VoiceOver.

1. Create some orders.
2. On the store, go to the My Account -> Orders.
3. Turn on the screen reader.
4. Navigate through the table cells. On VoiceOver, pressing "Control + Option + Arrow keys" makes this possible.
5. When you jump to a new row for the first time, the content of the first cell in the row should be announced. This is because the first cell has been turned into the header of the row.
6. On VoiceOver, When you jump to the first row, where all the headers leave, the content of the first cell in the second row is also announced. This is due to the change of the HTML tag of this cell from `<td>` to `<th>`. The same behavior can be found on the W3C example of an accessible table: https://www.w3.org/WAI/tutorials/tables/two-headers/ 

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [x] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

**Comment by Manussakis** [comment]
@dkotter This task is ready for code review.

File: plugins/woocommerce/templates/myaccount/orders.php [hunk 1]
```diff
@@ -39,12 +39,17 @@
...
 				?>
 				<tr class="woocommerce-orders-table__row woocommerce-orders-table__row--status-<?php echo esc_attr( $order->get_status() ); ?> order">
-					<?php foreach ( wc_get_account_orders_columns() as $column_id => $column_name ) : ?>
-						<td class="woocommerce-orders-table__cell woocommerce-orders-table__cell-<?php echo esc_attr( $column_id ); ?>" data-title="<?php echo esc_attr( $column_name ); ?>">
+					<?php foreach ( wc_get_account_orders_columns() as $column_id => $column_name ) :
+						$is_order_number = 'order-number' === $column_id;
+						$cell_html_tag   = $is_order_number ? 'th scope="row"' : 'td';
+						$cell_html_attrs = 'class="woocommerce-orders-table__cell woocommerce-orders-table__cell-' . esc_attr( $column_id ) . '"' . ' ' . 'data-title="' . esc_attr( $column_name ) . '"';
+					?>
+						<?php echo '<' . $cell_html_tag . ' ' . $cell_html_attrs . '>'; ?>
```
Comment (dkotter) [inline plugins/woocommerce/templates/myaccount/orders.php]: This works but is not ideal to directly echo this without any escaping. While it's safe right now as this value will always either be a `th` or `td`, if this code ever changes in the future, we may not be sure of that.

Wondering if instead we just have a conditional here that outputs the right thing, similar to the line added below: `<?php echo $is_order_number ? '</th>' : '</td>'; ?>`

Something like:
```php
<?php if ( $is_order_number ) : ?>
    <th scope="row" ATTRS HERE>
<?php else : ?>
    <td ATTRS HERE>
<?php endif; ?>
```

**Review by dkotter — CHANGES_REQUESTED** [review]

File: plugins/woocommerce/templates/myaccount/orders.php (hunk 1 above)
Comment (Manussakis) [reply plugins/woocommerce/templates/myaccount/orders.php]: Thanks for the feedback, @dkotter !
Please review the changes: [Refactor how order cell is rendered](https://github.com/woocommerce/woocommerce/pull/49201/commits/14cc1565edf28e3d75985db910039134654ebb03)

**Review by dkotter — APPROVED** [review]
@qasumitbagthariya @ankitguptaindia This is ready for QA

**Comment by ankitguptaindia** [comment]
**QA/Test Report-** ✅ 
------------

**Testing Environment -** 

<Details>

* WordPress: 6.5.5
* Theme: Twenty Twenty-Four Version: 1.1
* WooCommerce - Dev version with this fix-specific branch.  
* PHP: 8.3
* Web Server: Nginx
* Browser: Chrome - Version 126
* OS: macOS Sonoma 14.5
* Git Branch: Manussakis:fix/43629


</Details>


**Test Results -** 

Changes work as expected. Tested via macOS Voice Over with Twenty-Twenty-Four theme
 

**Functional Demo /  Screencast -**

**With macOS Voice Over:**


https://github.com/woocommerce/woocommerce/assets/6384299/049745ff-bc0b-42bc-aa4e-b7fde12971ef

**Comment by opr** [comment]
👋🏼  Are you folks expecting a review from Rubik on this one? Please let me know if so. When you are, please change this PR to `Ready for review` and I'll take a look.

**Comment by vikrampm1** [comment]
@Manussakis please remove WIP from the title and make the PR ready for review.

@opr Yes, this is ready for review and will be moved out of draft.

**Review by opr — APPROVED** [review]
Thanks @Manussakis this looks good, tests are passing too, and I think the failing checks are just because this is a fork.

**Pull request merged**